renderdoc/os/win32/win32_shellext.cpp

```diff
@@ -121,6 +124,39 @@ struct RDCThumbnailProvider : public IThumbnailProvider, IInitializeWithStream
 
     delete[] buf;
 
+    if(is_dds_file(captureHeader.data(), numRead))
+    {
+      // Read File from beginning. Seek requires the struct LARGE_INTEGER.
+      LARGE_INTEGER zero;
+      zero.QuadPart = 0;
+      pstream->Seek(zero, STREAM_SEEK_SET, NULL);
+      STATSTG stats = {};
+      pstream->Stat(&stats, STATFLAG_DEFAULT);
+      ULARGE_INTEGER size = stats.cbSize;
```

Comment:
Is there a reason we seek to the beginning here and re-read the start of the file that is already in `captureHeader`? Would it be possible to instead resize to expand up `captureHeader` to the full size and read the remaining part of the file at an offset? For files up to 2MB this would mean no extra work too, which is nice.

Reply:
Yes indeed I should have thought of that :)
Now `captureHeader` is resized to the file size, if the file is larger than 2MB, and then the rest of the file is read into the buffer.